### PR TITLE
Customer view

### PR DESCRIPTION
**Pull request by sebastianbot6969**

### CODE REVIEW

File: Flea/Pages/CustomerOverview.razor [hunk 1]
```diff
@@ -1,53 +1,53 @@
 @page "/CustomerOverview"
-@using System.Data.Common
+@using Customer = Flea.Models.Customer
 <div>
```
Comment (pimber) [inline Flea/Pages/CustomerOverview.razor:4]: Husk at smide en enter så der er linje afstand. Formatering :)

File: Flea/Pages/CustomerOverview.razor [hunk 2]
```diff
@@ -1,53 +1,53 @@
...
 @code
 {
+    private Modal _child;
+    private Customer _temp = new("", "");
     private string _search = "";
 
-    private List<Models.Customer> _customers = new()
+    // Dummy data ---
+
+    private List<Customer> Customers = new()
```
Comment (jonasKjellerup) [inline Flea/Pages/CustomerOverview.razor]: private field `Customers` does not adhere to naming conventions

File: Flea/Pages/CustomerOverview.razor [hunk 3]
```diff
@@ -1,53 +1,53 @@
...
-        new Models.Customer("Hans Nielsen", "8888888") {Id = 3},
+        new Customer("Sebastian Aaholm", "24823423") {Id = 1},
+        new Customer("Jens Jensen", "4541341") {Id = 2},
+        new Customer("Hans Nielsen", "8888888") {Id = 3}
     };
+
     // ---
 
-    private IEnumerable<Models.Customer> _shownCustomers;
+    private IEnumerable<Customer> ShownCustomers;
```
Comment (jonasKjellerup) [inline Flea/Pages/CustomerOverview.razor]: private field `ShownCustomers` does not adhere to naming conventions

File: Flea/Pages/CustomerOverview.razor [hunk 4]
```diff
@@ -1,53 +1,53 @@
...
-        <input class="search-input" type="search" value="" placeholder="Søg efter kunde" @oninput="Searchbar">
-        <button class="button button--outline create-button" @onclick="NewCustomer">Tilføj Ny Loppe</button>
+        <input class="search-input" type="search" value="" placeholder="Søg efter loppe" @oninput="Searchbar">
+        <Modal ButtonName="Opret ny loppe" @ref="_child">
+            <CustomerForm Customer="_temp" OnValidSubmit="NewCustomer" Modal="_child"></CustomerForm>
+        </Modal>
     </div>
-
-    @foreach (var customer in _shownCustomers)
+    @foreach (var customer in ShownCustomers)
```
Comment (pimber) [inline Flea/Pages/CustomerOverview.razor]: _showCustomers

File: Flea/Pages/CustomerOverview.razor (hunk 3 above)
Comment (pimber) [inline Flea/Pages/CustomerOverview.razor]: _showCustomer

File: Flea/Pages/CustomerOverview.razor [hunk 5]
```diff
@@ -1,53 +1,53 @@
...
-    private IEnumerable<Models.Customer> _shownCustomers;
+    private IEnumerable<Customer> ShownCustomers;
 
     // Delete the customer
     public void DeleteCustomerInformation(int id)
     {
-        _customers = _customers.Where(c => c.Id != id).ToList();
-        _shownCustomers = _shownCustomers.Where(c => c.Id != id);
+        Customers = Customers.Where(c => c.Id != id).ToList();
+        ShownCustomers = ShownCustomers.Where(c => c.Id != id);
```
Comment (pimber) [inline Flea/Pages/CustomerOverview.razor]: Kig på name convention

File: Flea/Pages/CustomerOverview.razor [hunk 6]
```diff
@@ -57,31 +57,31 @@
         var input = e.Value?.ToString() ?? "";
         if (input == "")
         {
-            _shownCustomers = _customers;
+            ShownCustomers = Customers;
```
Comment (pimber) [inline Flea/Pages/CustomerOverview.razor]: nameing

File: Flea/Pages/CustomerOverview.razor [hunk 7]
```diff
@@ -57,31 +57,31 @@
...
         }
         else if (Char.IsDigit(input[0]))
         {
     // Finds by phonenumbers
-            
+
             var results = from c
-                in _customers
+                in Customers
```
Comment (pimber) [inline Flea/Pages/CustomerOverview.razor]: naming

File: Flea/Pages/CustomerOverview.razor [hunk 8]
```diff
@@ -57,31 +57,31 @@
...
                 select c;
-            _shownCustomers = results;
+            ShownCustomers = results;
         }
         else
         {
     // Finds by name
             var results = from c
-                in _customers
+                in Customers
```
Comment (pimber) [inline Flea/Pages/CustomerOverview.razor]: naming

File: Flea/Pages/CustomerOverview.razor [hunk 9]
```diff
@@ -57,31 +57,31 @@
...
                 where c.Name.ToLower().Contains(input.ToLower())
                 select c;
-            _shownCustomers = results;
+            ShownCustomers = results;
```
Comment (pimber) [inline Flea/Pages/CustomerOverview.razor]: naming

File: Flea/Pages/CustomerOverview.razor [hunk 10]
```diff
@@ -57,31 +57,31 @@
...
         }
     }
 
     protected override void OnInitialized()
     {
-        _shownCustomers = _customers;
+        ShownCustomers = Customers;
```
Comment (pimber) [inline Flea/Pages/CustomerOverview.razor]: nameing

File: Flea/Shared/Modal.razor [hunk 11]
```diff
@@ -1,9 +1,9 @@
 <button class="button button--outline create-button" @onclick="@ModalShow">@ButtonName</button>
 @if (_showModal)
 {
-    <div class="modal fade show" id="myModal" style="display:block; background-color: transparent;" aria-modal="true" role="dialog">
+    <div class="modal fade show" id="myModal" style="display: block; background-color: transparent" aria-modal="true" role="dialog">
```
Comment (pimber) [inline Flea/Shared/Modal.razor]: Inline style skal laves som en class og smides på class="modal fade show <et eller andet navn>"

File: Flea/Shared/Modal.razor [hunk 12]
```diff
@@ -1,9 +1,9 @@
...
         <div class="modal-dialog">
-            <div class="modal-content">
+            <div class="modal-content" >
```
Comment (pimber) [inline Flea/Shared/Modal.razor]: Fjern mellemrum

File: Flea/Shared/Modal.razor [hunk 13]
```diff
@@ -16,8 +16,8 @@
     
     [Parameter]
     public string ButtonName { get; set; }
-    
+
     private bool _showModal = false;
-    private void ModalShow() => _showModal = true;
+    private void ModalShow()=> _showModal = true;
```
Comment (pimber) [inline Flea/Shared/Modal.razor]: mellemrum

File: Flea/Shared/CustomerUserListItem.razor [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+    <div class="txt">Navn: @Customer.Name</div>
+    <div class="txt">Nummer: @Customer.PhoneNumber</div>
+    <button class="button button--outline del-button" @onclick="Delete">Slet</button>
+    <Modal ButtonName="Rediger" @ref="_child">
+        <CustomerForm Customer="_temp" OnValidSubmit="OnSubmit" Modal="_child"></CustomerForm>
+    </Modal>
+</div>
+
+@code {
+    Modal _child;
```
Comment (jonasKjellerup) [inline Flea/Shared/CustomerUserListItem.razor:15]: should be marked as `private`

**Review by jonasKjellerup — CHANGES_REQUESTED** [review]

**Review by jonasKjellerup — APPROVED** [review]

File: Flea/Shared/CustomerForm.razor [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+        @* <div> *@
+        @*     <label>Leje af tøj stativ: </label> *@
+        @*     <InputNumber @bind-Value="Temp.Comments"></InputNumber> *@
+        @* </div> *@
+        @* <div> *@
+        @*     <label>Bemærkninger: </label> *@
+        @*     <InputText @bind-Value="Temp.Comments"></InputText> *@
+        @* </div> *@
+        <div id="move-down">
+            <button type="submit" class="button create-button button--outline">@(Customer.Id == 0 ? "Opret Kunde" : "Gem ændringer")</button>
```
Comment (pimber) [inline Flea/Shared/CustomerForm.razor]: Opret loppe

**Review by pimber — APPROVED** [review]

**Pull request merged**